File: SunnyFlamingo/ClientApp/src/app/models/order-model.ts
import { BasketGoodModel } from "./basket-good-model";
import { DeliveryMethodModel } from "./delivery-method-model";
import { UserModel } from "./user-model";

export class OrderModel{
  id?: string;
  address: string;
  deliveryMethod: DeliveryMethodModel;
  user: UserModel;
  basketGoods: BasketGoodModel[];
  constructor(
    options: {
    id?: string, 
    address: string, 
    deliveryMethod: DeliveryMethodModel, 
    userModel: UserModel, 
    basketGoods: BasketGoodModel[]}) {
    this.id = options.id;
    this.address = options.address;
    this.deliveryMethod = options.deliveryMethod;
    this.user = options.userModel;
    this.basketGoods = options.basketGoods;
  }
}

File: SunnyFlamingo/Models/OrderModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class OrderModel
    {
        public Guid Id { get; set; }
        public string Address { get; set; }


        public DeliveryMethodModel DeliveryMethod { get; set; }
        public UserModel User { get; set; }
        public List<BasketGoodModel> BasketGoods { get; set; }
    }
}
